src/lib.rs

```diff
@@ -711,15 +712,14 @@ pub mod tests {
         }
         println!("Generated query set");
 
-        let opening_challenge = F::rand(rng);
         let proof = PC::batch_open(
             &ck,
             &polynomials,
             &comms,
             &query_set,
-            opening_challenge,
             &rands,
             Some(rng),
+            &mut PC::setup_oracle(),
```

Comment:
Ah I understand now why you needed this function: but also in this case, I would replace the "opening_challenge" parameter (that you deleted) with a random oracle, and just forward it (like you did in the functions above btw, is this a leftover ?)

Reply:
Done